### ISSUE TITLE
eb deploy - [Errno 102] Operation not supported on socket

### ISSUE THREAD

**Issue by jimryan** [aws/aws-elastic-beanstalk-cli#22]
### Description

When attempting to deploy an application that has a socket file anywhere in it, an error is returned and the deploy fails. Even if the file is ignored in `.ebignore`.

In my specific case,  I'm attempting to deploy a Ruby on Rails application on a machine using [puma-dev](https://github.com/puma/puma-dev), and see the following error:

```
ERROR: OSError - [Errno 102] Operation not supported on socket: './tmp/puma-dev-7584.sock'
```

When I stop puma-dev (thus removing the `.sock` file), `eb deploy` works.

I have `/tmp/*` in my `.ebignore` file, and can confirm that file is generally working as other ignored files are not deployed.

I can also confirm, unsurprisingly, that this occurs with _any_ socket file, and is not an issue unique to puma-dev. 

### Steps to reproduce

Here are steps that do not involve puma-dev to make it easier to reproduce:

1. `cd` into your project directory
2. Create a socket file, e.g.:
    ```
    python -c "import socket as s; sock = s.socket(s.AF_UNIX); sock.bind('test.sock')"
    ```
3. Ignore the socket (this isn't necessarily required, but my expectation is that the cli would at the very least ignore sockets that are explicitly ignored in `.ebignore`, perhaps debatable if they should just always be ignored)
    ```
    echo "test.sock" >> .ebignore
    ```
4. Attempt to deploy
    ```
    eb deploy
    ```

### Observed result

Only including last bit for brevity, but happy to include whole thing if needed:

```
2020-04-02 14:46:34,552 (INFO) eb : Found .ebignore, using system zip.
2020-04-02 14:46:34,553 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/jim/Personal Projects/app
2020-04-02 14:46:34,553 (INFO) eb : Zipping up folder at location: /Users/jim/Personal Projects/app
2020-04-02 14:46:34,563 (INFO) eb :  +adding: .//
2020-04-02 14:46:34,564 (INFO) eb :   +adding: ./.rspec
2020-04-02 14:46:34,565 (INFO) eb :   +adding: ./.DS_Store
2020-04-02 14:46:34,566 (INFO) eb :   +adding: ./.ruby-version
2020-04-02 14:46:34,566 (INFO) eb :   -skipping: ./yarn-error.log
2020-04-02 14:46:34,567 (INFO) eb :   +adding: ./config.ru
2020-04-02 14:46:34,567 (INFO) eb :   +adding: ./test.sock
2020-04-02 14:46:34,575 (INFO) eb : Traceback (most recent call last):
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/core/ebrun.py", line 62, in run_app
    app.run()
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/cement/core/foundation.py", line 797, in run
    return_val = self.controller._dispatch()
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/cement/core/controller.py", line 478, in _dispatch
    return func()
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/core/abstractcontroller.py", line 92, in default
    self.do_command()
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/controllers/deploy.py", line 76, in do_command
    deployops.deploy(self.app_name, self.env_name, self.version, self.label,
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/operations/deployops.py", line 53, in deploy
    app_version_label = commonops.create_app_version(
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/operations/commonops.py", line 528, in create_app_version
    file_name, file_path = _zip_up_project(
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/operations/commonops.py", line 723, in _zip_up_project
    fileoperations.zip_up_project(file_path, ignore_list=ignore_files)
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/core/fileoperations.py", line 420, in zip_up_project
    zip_up_folder('./', location, ignore_list=ignore_list)
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/core/fileoperations.py", line 407, in zip_up_folder
    _zipdir('./', zipf, ignore_list=ignore_list)
  File "/usr/local/Cellar/aws-elasticbeanstalk/3.17.1_1/libexec/lib/python3.8/site-packages/ebcli/core/fileoperations.py", line 473, in _zipdir
    zipf.write(cur_file)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/zipfile.py", line 1775, in write
    with open(filename, "rb") as src, self.open(zinfo, 'w') as dest:
OSError: [Errno 102] Operation not supported on socket: './test.sock'

2020-04-02 14:46:34,575 (INFO) eb : OSError - [Errno 102] Operation not supported on socket: './test.sock'
```

### Expected result

CLI ignores the socket file, either because it _always_ ignores files that it cannot deploy, or at least because the socket file is explicitly ignored in `.ebignore`. It may make sense to not ignore sockets unless they are ignored in `.ebignore` to alert the user to a file that cannot be deployed, and their remedy can be to `.ebignore` it, but if it is ignored, it should definitely not raise this error and stop the deploy.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS 10.14.4

Thanks - be safe and be well.

**Comment by halcarleton** [comment]
This should be fixed in [v3.18.2](https://pypi.org/project/awsebcli/3.18.2/)

**Comment by jimryan** [comment]
Thanks @halcarleton!
